src/functions/src/main/java/org/apache/jmeter/functions/DigestEncodeFunction.java

```diff
@@ -103,12 +100,9 @@ public String execute(SampleResult previousResult, Sampler currentSampler) throw
      * @return
      */
     private String uppercase(String encodedString, CompoundVariable[] values, int index) {
-        if (values.length > index) {
-            String shouldUpperCase = values[index].execute();
-            boolean shouldDoUpperCase = Boolean.TRUE.toString().equalsIgnoreCase(shouldUpperCase);
-            if (shouldDoUpperCase) {
-                return encodedString.toUpperCase();
-            }
+        String shouldUpperCase = values.length > index ? values[index].execute() : null;
+        if (Boolean.TRUE.toString().equalsIgnoreCase(shouldUpperCase)) {
```

Comment:
Is this `Boolean.parseBoolean(shouldUpperCase)`?

Reply:
@vlsi yes, created a new Pull

Reply:
Why do you create a new one? Would you please just update the current PR?